docs/design/CONFIG_API.md

```diff
@@ -0,0 +1,231 @@
+- Build and publish a container from a Dockerfile.
+
+After installing the latest Kubebuilder release, create the `BootstrapConfig` API:
+```shell
+kubebuilder create api --group envoygateway.io --version v1alpha1 --kind BootstrapConfig
+```
+
+The `v1alpha1` version and `envoygateway.io` API group get generated:
+```go
+// gateway/api/v1alpha1/doc.go
+
+// Package v1alpha1 contains API Schema definitions for the envoygateway.io API group.
+//
+// +groupName=config.envoygateway.io
+package v1alpha1
+```
+
+The initial `BootstrapConfig` API being proposed:
+```go
+// gateway/api/v1alpha1/bootstrapconfig_types.go
+
+// BootstrapConfigSpec defines the desired state of BootstrapConfig.
+type BootstrapConfigSpec struct {
+	// EnabledServices defines the services to enable. If unspecified,
+	// defaults to enabling all supported services.
+	//
+	// +optional
+	// +kubebuilder:validation:MaxItems=12
+	// +kubebuilder:validation:Enum=XdsServer;Provisioner
+	// +kubebuilder:default=XdsServer;Provisioner
+	EnabledServices []ServiceType `json:"enabledServices,omitempty"`
+
+	// XDSServer contains parameters of the xDS server. If unspecified,
+	// the xDS server will use default configuration settings.
+	//
+	// +optional
+	XDSServer *XDSServerConfig `json:"xdsServer,omitempty"`
+
+	// Provisioner contains parameters of the provisioner server. If
+	// unspecified, the provisioner server will use default configuration
+	// settings.
+	//
+	// +optional
+	Provisioner *ProvisionerConfig `json:"Provisioner,omitempty"`
+}
+
+// ServiceType defines the type of services supported by Envoy Gateway.
+//
+// +kubebuilder:validation:Enum=XdsServer;Provisioner
+type ServiceType string
+
+const (
+	// XdsServerServiceType is the name of the "XdsServer" service type.
+    XdsServerServiceType ServiceType = "XdsServer"
+
+	// ProvisionerServiceType is the name of the "Provisioner" service type.
+    ProvisionerServiceType ServiceType = "Provisioner"
+)
+
+// XDSServerConfig contains the configuration of the xDS server.
+type XDSServerConfig struct {
+	// Defines the IP address of the server to serve xDS over gRPC.
+	//
+	// If unspecified, defaults to "0.0.0.0", e.g. all IP addresses.
+	//
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:default=0.0.0.0
+	Address *string `json:"address,omitempty"`
+
+	// Defines the network port for the xDS server to serve xDS over gRPC.
+	//
+	// If unspecified, defaults is 18000.
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=65535
+	// +kubebuilder:default=18000
+	Port *int32 `json:"port,omitempty"`
+}
+
+// ProvisionerConfig defines the provisioner server configuration.
+type ProvisionerConfig struct {
+	// ControlPlane specifies deployment configuration of the control plane,
+	// e.g. the xDS server.
+	//
+	// +optional
+	ControlPlane *ControlPlaneConfig `json:"controlPlane,omitempty"`
```

Comment:
we haven't designed https://github.com/envoyproxy/gateway/issues/40 yet but based on the comments [here](https://github.com/envoyproxy/gateway/issues/60#issuecomment-1143637841), it is possible that there would be a single xDS server for all the data planes so this field might not be required here